src/pubsub/src/publisher/client.rs

```diff
@@ -159,6 +157,60 @@ impl Publisher {
     }
 }
 
+#[derive(Clone, Debug)]
+pub struct PublisherBuilder {
+    pub(crate) inner: crate::generated::gapic_dataplane::client::Publisher,
+    topic: String,
+    batching_settings: BatchingSettings,
+}
+
+impl PublisherBuilder {
+    /// Creates a new Pub/Sub publisher builder for topic.
+    pub(crate) fn new(
+        client: crate::generated::gapic_dataplane::client::Publisher,
+        topic: String,
+    ) -> Self {
+        Self {
+            inner: client,
+            topic,
+            batching_settings: BatchingSettings::default(),
+        }
+    }
+
+    // Change the message batching settings.
+    pub fn with_batching(mut self, settings: BatchingSettings) -> PublisherBuilder {
```

Comment:
tests for this?

we could have the field be `pub(crate)` on the client and inspect it in a test.